model/src/app_data.rs

```diff
@@ -0,0 +1,132 @@
+//! Contains the app_data file structures to define additional data about tx origin
+
+use crate::h160_hexadecimal::{self};
+use anyhow::{anyhow, Result};
+use cid::multihash::{Code, MultihashDigest};
+use primitive_types::{H160, H256};
+use serde::{Deserialize, Serialize};
+use serde_with::serde_as;
+use std::convert::TryInto;
+
+#[serde_as]
+#[derive(Eq, PartialEq, Clone, Debug, Deserialize, Serialize, Hash, Default)]
+pub struct Referrer {
+    #[serde(with = "h160_hexadecimal")]
+    pub referrer: H160,
```

Comment:
maybe this can be called `address` to address @leafilipo's concern with the naming?

Reply:
Wrong tag, but I agree with the change.
cc @anxolin for visibility